scripts/dependencies/lima.ts

```diff
@@ -244,8 +248,24 @@ export class AlpineLimaISO implements Dependency, GitHubDependency {
   }
 
   rcompareVersions(version1: AlpineLimaISOVersion, version2: AlpineLimaISOVersion): -1 | 0 | 1 {
-    const semverVersion1 = this.versionToSemver(version1);
-    const semverVersion2 = this.versionToSemver(version2);
+    let semverVersion1 = '0.0.0';
+    let invalidVersion1 = false;
+    let semverVersion2 = '0.0.0';
+    let invalidVersion2 = false;
+
+    try {
+      semverVersion1 = this.versionToSemver(version1);
+    } catch (ex) {
+      invalidVersion1 = true;
+    }
+    try {
+      semverVersion2 = this.versionToSemver(version2);
+    } catch (ex) {
+      invalidVersion2 = true;
+    }
+    if (invalidVersion1 || invalidVersion2) {
+      return !invalidVersion1 ? -1 : invalidVersion2 ? 1 : 0;
```

Comment:
I may be mistaken, but I think there may be an error here.

If version 1 is invalid and version 2 is invalid, then we return 1. Shouldn't we return 0?

If version 1 is invalid and version 2 is valid, then we return 0. Shouldn't we return 1?

Maybe this would be better written as a `if-else` chain? I'm kind of tying my brain in knots, especially with the double negatives.

Reply:
I think you're right, and there's an error here (`invalidVersion2` should be inverted too); but I'll write this out longhand to be clearer.